src/_pytest/junitxml.py

```diff
@@ -186,9 +186,18 @@ def _prepare_content(self, content: str, header: str) -> str:
         return "\n".join([header.center(80, "-"), content, ""])
 
     def _write_content(self, report: TestReport, content: str, jheader: str) -> None:
-        tag = ET.Element(jheader)
-        tag.text = bin_xml_escape(content)
-        self.append(tag)
+        exists = False
+        for elem in next(iter(self.xml.node_reporters.values())).nodes:
+            if jheader in elem.tag:
+                exists = True
+                tag = elem
+                tag.text = (
+                    tag.text + bin_xml_escape(content) if tag.text is not None else None
+                )
+        if not exists:
+            tag = ET.Element(jheader)
+            tag.text = bin_xml_escape(content)
+            self.append(tag)
```

Comment:
`else` is great for such a case (also seems like that `else None` is incorrect):

```suggestion
        for elem in next(iter(self.xml.node_reporters.values())).nodes:
            if elem.tag == jheader:
                text = elem.text if elem.text is not None else ""
                elem.text = text + "\n" + bin_xml_escape(content)
                break
        else:
            tag = ET.Element(jheader)
            tag.text = bin_xml_escape(content)
            self.append(tag)
```

Reply:
Another question, why that `next` is needed`? Shouldn't you just iterate over all reporters (can't look at the code to make sure, hence the question).